crates/uv/tests/it/pip_compile.rs

```diff
@@ -17403,3 +17403,45 @@ fn compile_broken_active_venv() -> Result<()> {
 
     Ok(())
 }
+
+/// <https://github.com/astral-sh/uv/issues/13344>
+#[test]
+fn pubgrub_panic_twice_self_dep() -> Result<()> {
+    let context = TestContext::new("3.12");
+    context
+        .temp_dir
+        .child("pyproject.toml")
+        .write_str(indoc! {r#"
+        [project]
+        name = "foo"
+        version = "0.1.0"
+        dependencies = [
+            "foo",
+            "foo",
```

Comment:
just for completeness sake, can you imagine a scenario where you have `foo` and a double self dependency `foo[extra-bar]` that also has an extra with it?